django/transformer/reference_binder.py

```diff
@@ -0,0 +1,103 @@
+import logging
+from typing import List
+
+from dotty_dict import dotty
+from loader.load.utils import get_resource_id
+from transformer.errors import IncompleteIdentifierError
+from transformer.transform.transformer import compute_fhir_object_id
+
+logger = logging.getLogger(__name__)
+
+
+class ReferenceBinder:
+    def resolve_references(self, unresolved_fhir_object, reference_paths: List[List[str]]):
+        fhir_object = dotty(unresolved_fhir_object)
+        resource_id = get_resource_id(unresolved_fhir_object)
+
+        # iterate over the instance's references and try to resolve them
+        for reference_path in reference_paths:
+            logger.debug(
+                {
+                    "message": f"Trying to resolve reference for resource {fhir_object['id']} "
+                    f"at {'[*]'.join(reference_path)}",
+                    "resource_id": resource_id,
+                },
+            )
+            try:
+                self.bind_existing_reference(fhir_object, reference_path)
+            except KeyError as e:
+                logger.warning(f"{reference_path} does not exist in resource {fhir_object['id']}: {e}")
+
+        return fhir_object.to_dict()
+
+    def bind_existing_reference(self, fhir_object, reference_path: List[str]):
```

Comment:
`reference_path` should be plural `reference_paths`

Reply:
Nope, we're looping on all the reference_paths one by one

Reply:
sorry you're right 🙃